airbyte-db/lib/src/main/java/io/airbyte/db/AbstractDatabase.java

```diff
@@ -6,7 +6,7 @@
 
 import com.fasterxml.jackson.databind.JsonNode;
 
-public abstract class AbstractDatabase implements AutoCloseable {
```

Comment:
why did you remove this from here?

Reply:
The database class is no longer what is closeable.  It is merely now a wrapper around the instantiated `DataSource`, which is what we actually want to close.  By allowing the `AbstractDatabse` to close the data source, we might accidentally close a shared data source between multiple instances.  Therefore, it is the data source that should be directly closed, removing the need for this class to implement closeable (it has nothing to close now that inversion of control has been enforced).

Reply:
This explanation is helpful. Maybe copy some of that as the Javadoc of this class? Something like:

```java
/**
 * The Database class is a wrapper around the instantiated DataSource. The DataSource is closeable, but
 * not the Database class. This is because we don't want to accidentally close a shared data source between
 * multiple instances by closing the Database class.
 */
```